contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,264 @@
+
+        emit ExecutionDelaySet(executionDelay);
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function getDisputableAction(uint256 _delayedScriptId)
+        external view returns (uint64 endDate, bool challenged, bool finished)
+    {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        DelayedScriptStatus delayedScriptStatus = delayedScript.delayedScriptStatus;
+
+        endDate = delayedScript.executionFromTime;
+        challenged = delayedScriptStatus == DelayedScriptStatus.Paused;
+        finished = delayedScriptStatus == DelayedScriptStatus.Cancelled
+            || delayedScriptStatus == DelayedScriptStatus.Executed;
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function canChallenge(uint256 _delayedScriptId) external view returns (bool) {
+        return _canPause(delayedScripts[_delayedScriptId]);
+    }
+
+    /**
+    * @dev IDisputable interface conformance
+    */
+    function canClose(uint256 _delayedScriptId) external view returns (bool) {
+        return _canClose(delayedScripts[_delayedScriptId]);
+    }
+
+    /**
+    * @notice Delays execution for `@transformTime(self.executionDelay(): uint)`
+    * @param _context Information context for the script being scheduled
+    * @param _evmCallScript The script that can be executed after a delay
+    */
+    function delayExecution(bytes _context, bytes _evmCallScript) external auth(DELAY_EXECUTION_ROLE) returns (uint256) {
+        return _delayExecution(_context, _evmCallScript);
+    }
+
+    /**
+    * @dev IForwarder interface conformance
+    */
+    function isForwarder() external pure returns (bool) {
+        return true;
+    }
+
+    /**
+    * @dev IForwarder interface conformance
+    */
+    function canForward(address _sender, bytes) public view returns (bool) {
+        return canPerform(_sender, DELAY_EXECUTION_ROLE, arr());
+    }
+
+    /**
+    * @notice Delays execution for `@transformTime(self.executionDelay(): uint)`
+    * @dev IForwarder interface conformance
+    * @param _evmCallScript The script that can be executed after a delay
+    */
+    function forward(bytes _evmCallScript) public {
+        require(canForward(msg.sender, _evmCallScript), ERROR_CANNOT_FORWARD);
+        _delayExecution(new bytes(0), _evmCallScript);
+    }
+
+    /**
+    * @dev Challenge script execution, IDisputable interface conformance
+    * @param _delayedScriptId The ID of the script execution to challenge
+    */
+    function _onDisputableActionChallenged(uint256 _delayedScriptId, uint256 /* _challengeId */, address /* _challenger */)
+        internal
+    {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        delayedScript.pausedAt = getTimestamp64();
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Paused;
+
+        emit ExecutionPaused(_delayedScriptId, delayedScript.actionId);
```

Comment:
I think it does worth adding a comment here expressing we are trusting the Agreement app to make sure action will be challenged only when `canChallenge` returns true, which is part of the contract ofc.